src/pages/MainPage.tsx

```diff
@@ -7,51 +7,60 @@ import Banner from '../views/MainPage/components/Banner';
 import GuestContents from '../views/MainPage/components/GuestContents';
 import StatusBarForiOS from '../views/MainPage/components/StatusBarForiOS';
 import TopSheet from '../views/MainPage/components/TopSheet';
-import { ReceivedOffer, ReceivedApplication } from '../views/MainPage/components/UserContents';
+import { DesignerContents, ModelContents } from '../views/MainPage/components/UserContents';
 
 import { userTypeState } from '@/recoil/atoms/signUpState';
 import { USER_TYPE } from '@/views/@common/constants/userType';
+import { INITIAL_PAGE } from '@/views/MainPage/constants/constant';
 import useGetMain from '@/views/MainPage/hooks/useGetMain';
 
 const MainPage = () => {
   const userType = useRecoilValue(userTypeState);
+  const [page, setPage] = useState(INITIAL_PAGE);
+  const { modelData, designerData } = useGetMain({ user: userType, page: page });
 
   // 뒤로 가기 막기 관련
   const navigate = useNavigate();
-  window.history.pushState(null, '', '');
-  window.onpopstate = () => {
-    navigate('/');
-  };
+  useEffect(() => {
+    const handlePopState = () => navigate('/');
+    window.history.pushState(null, '', window.location.href);
+    window.addEventListener('popstate', handlePopState);
+
+    return () => window.removeEventListener('popstate', handlePopState);
+  }, [navigate]);
 
-  const [page, setPage] = useState(1);
-  const { data } = useGetMain({ user: userType, page: page });
+  const Contents = {
+    [USER_TYPE.GUEST]: {
+      mainContent: GuestContents,
+    },
+    [USER_TYPE.DESIGNER]: {
+      mainContent: () => designerData && <DesignerContents data={designerData} setPage={setPage} />,
+      name: designerData?.name,
+    },
+    [USER_TYPE.MODEL]: {
+      mainContent: () => modelData && <ModelContents data={modelData} setPage={setPage} />,
+      applyType: modelData?.status,
+      name: modelData?.name,
+    },
+  };
 
-  const MainContents = () => {
-    switch (userType) {
-      case USER_TYPE.GUEST:
-        return <GuestContents />;
-      case USER_TYPE.DESIGNER:
-        return data && <ReceivedApplication data={data} setPage={setPage} />;
-      case USER_TYPE.MODEL:
-        return data && <ReceivedOffer data={data} setPage={setPage} />;
-      default:
-        return null;
-    }
+  const TopSheetProps = {
+    userType,
+    applyType: Contents[userType].applyType,
+    name: Contents[userType].name,
   };
 
+  const MainContentsByUserType = () => {
+    const ContentComponent = Contents[userType].mainContent;
+    return <ContentComponent />;
+  };
```

Comment:
```suggestion
  const MainContentsByUserType = () => {
    const ContentComponent = Contents[userType].mainContent;
    return ContentComponent;
  };
```

위의 Contents를 수정하면 아래와 같이 간소화할 수 있게 됩니다.

```suggestion
  const MainContentsByUserType = () => Contents[userType].mainContent;
```

그리고 이렇게 간소화해보면, `MainContentsByUserType` 컴포넌트의 존재 의미가 사라지게 됩니다. 
`MainContentsByUserType`를 별도로 빼서 사용하신 수빈님의 의도가 만약
리턴문에서 **컴포넌트 형태로 통일**하고 싶었던거라면 위와 같은 간소화된 형태를 사용하시면 될 것 같고, 
만약 통일할 필요 없다! 하시면 아래 suggestion과 같이 MainContentsByUserType 분리하지 않고 바로 mainContent를 사용하셔도 될 것 같아요!

Reply:
맞아요 return 문을 깔끔하게 해주기 위해서 컴포넌트 형태로 통일하는 것을 선호합니다!!
`<StatusBarForiOS />` 도 같은 이유로 컴포넌트 형태로 만들어준거거든요 ㅎㅎ